tests/sentry/workflow_engine/processors/test_workflow.py

```diff
@@ -400,3 +410,86 @@ def test_enqueue_workflow__adds_to_workflow_engine_set(
                 {"event_id": self.event.event_id, "occurrence_id": self.group_event.occurrence_id}
             ),
         )
+
+
+@django_db_all
+class TestDeleteWorkflow:
+    @pytest.fixture(autouse=True)
+    def setUp(self):
+        self.organization = Factories.create_organization()
+        self.project = Factories.create_project(organization=self.organization)
+
+        self.workflow = Factories.create_workflow()
+        self.workflow_trigger = Factories.create_data_condition_group(
+            organization=self.organization
+        )
+        self.workflow.when_condition_group = self.workflow_trigger
+        self.workflow.save()
+
+        self.action_filter = Factories.create_data_condition_group(organization=self.organization)
+        self.action = Factories.create_action()
+        self.action_and_filter = Factories.create_data_condition_group_action(
+            condition_group=self.action_filter,
+            action=self.action,
+        )
+
+        self.workflow_actions = Factories.create_workflow_data_condition_group(
+            workflow=self.workflow,
+            condition_group=self.action_filter,
+        )
+
+        self.trigger_condition = Factories.create_data_condition(
+            condition_group=self.workflow_trigger,
+            comparison=1,
+            condition_result=True,
+        )
+
+        self.action_condition = Factories.create_data_condition(
+            condition_group=self.action_filter,
+            comparison=1,
+            condition_result=True,
+        )
+
+    @pytest.mark.parametrize(
+        "instance_attr",
+        [
+            "workflow",
+            "workflow_trigger",
+            "action_filter",
+            "action_and_filter",
+            "workflow_actions",
+            "trigger_condition",
+            "action_condition",
+        ],
+    )
+    def test_delete_workflow(self, instance_attr):
+        instance = getattr(self, instance_attr)
+        instance_id = instance.id
+        cls = instance.__class__
+
+        delete_workflow(self.workflow)
+
+        # Checks that all the models are deleted in individual test cases
+        assert not cls.objects.filter(id=instance_id).exists()
+
+    def test_delete_workflow__no_actions(self):
+        # Remove the assiaction action and reference to the action to the action conditions
```

Comment:
what is `assiaction`?

Reply:
a keyboard not wanting to take `association` as my input? 🤞